zerver/views/message_fetch.py

```diff
@@ -791,6 +804,9 @@ def parse_anchor_value(anchor_val: Optional[str],
     except ValueError:
         raise JsonableError(_("Invalid anchor"))
 
+def get_realm_from_request(request: HttpRequest) -> Realm:
+    return Realm.objects.get(string_id=get_subdomain(request))
```

Comment:
We should use `get_realm_from_request` from `zerver/context_processors.py` instead of adding a simplified duplicate.